aws/resource_aws_instance_test.go

```diff
@@ -107,6 +107,51 @@ func TestAccAWSInstance_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSInstance_userDataBase64(t *testing.T) {
+	var v ec2.Instance
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { testAccPreCheck(t) },
+
+		// We ignore security groups because even with EC2 classic
+		// we'll import as VPC security groups, which is fine. We verify
+		// VPC security group import in other tests
+		IDRefreshName:   "aws_instance.foo",
+		IDRefreshIgnore: []string{"security_groups", "vpc_security_group_ids"},
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccInstanceConfigWithUserDataBase64,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInstanceExists(
+						"aws_instance.foo", &v),
+					resource.TestCheckResourceAttr(
+						"aws_instance.foo",
+						"user_data_base64",
+						"aGVsbG8gd29ybGQ="),
+				),
+			},
+
+			// We repeat the exact same test so that we can be sure
+			// that the handling of the user_data and user_data_base64
+			// distinction doesn't cause diff weirdness.
```

Comment:
Ahh yes, of course. This was something I inherited and tweaked from another nearby test, but I forgot that it's not necessary here. (The other test was doing it specifically to test some of the weird behavior we do to prevent double-base64-encoding, which isn't true here.)